precondition.go

```diff
@@ -14,7 +14,7 @@ import (
 var ErrPreconditionFailed = errors.New("task: precondition not met")
 
 func (e *Executor) areTaskPreconditionsMet(ctx context.Context, t *ast.Task) (bool, error) {
-	for _, p := range t.Preconditions {
+	for _, p := range append(t.Preconditions, e.Taskfile.Preconditions.Preconditions...) {
```

Comment:
I would expect the global `preconditions` to run first. For example:

```yaml
version: '3'

preconditions:
  - test -f foo.txt

tasks:
  default:
    preconditions:
      - test -f bar.txt
    cmds:
      - echo "default task"
```

Assuming that neither file exists, I would expect to see

```
task: `test -f foo.txt` failed
task: precondition not met
```

Also prefer `slices.Concat` over `append` and `...` now that its in the std lib.

```suggestion
	for _, p := range slices.Concat(e.Taskfile.Preconditions.Preconditions, t.Preconditions) {
```

Reply:
I agree
thanks for the suggestion